sources/metrics/detectors/ad progress/AdProgressDetectorTest.swift

```diff
@@ -0,0 +1,86 @@
+//  Copyright 2019, Oath Inc.
+//  Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
+import XCTest
+@testable import VerizonVideoPartnerSDK
+@testable import PlayerCore
+class AdProgressDetectorTest: XCTestCase {
+    var sut: Detectors.AdProgressDetector!
+    var timeProgress: PlayerCore.AdPixels.Progress!
+    var percentageProgress: PlayerCore.AdPixels.Progress!
+    
+    override func setUp() {
+        super.setUp()
+        
+        sut = Detectors.AdProgressDetector()
+        timeProgress = .init(url: testUrl, offset: .time(30))
+        percentageProgress = .init(url: testUrl, offset: .percentage(50))
+    }
+    
+    func testProcessNoOffset() {
```

Comment:
Also, nice to have a test case for detecting same time

Reply:
Same time?

Reply:
if we get current time same as we've got on the previous step

Reply:
Well, we have similar logic - we get a value that is higher than offset. I use `>=`, not `==`. Anyway, I'm gonna move this logic in core